### PR TITLE
Option it

### PR DESCRIPTION
**Pull request by faje18**

### CODE REVIEW

File: src/SMC/SMCMain.cpp [hunk 1]
```diff
@@ -56,7 +57,12 @@ namespace SMC
         }
 
         auto stubset = std::make_shared<SMCStubbornSet>(*net, query);
-        stubset->SMC::SMCStubbornSet::setInterestingVisitor<PetriEngine::InterestingSMCTransitionVisitor>();
+        if(aphi){
+            stubset->SMC::SMCStubbornSet::setInterestingVisitor<PetriEngine::InterestingTransitionVisitor>();
+        }
+        else{
+            stubset->SMC::SMCStubbornSet::setInterestingSMCVisitor<PetriEngine::InterestingSMCTransitionVisitor>();
```
Comment (fulei345) [inline src/SMC/SMCMain.cpp]:
```suggestion
        if(aphi){
            stubset->SMC::SMCStubbornSet::setInterestingSMCVisitor<PetriEngine::InterestingSMCTransitionVisitor>();
        }
        else{
            stubset->SMC::SMCStubbornSet::setInterestingVisitor<PetriEngine::InterestingTransitionVisitor>();
```

File: src/SMC/Stubborn/SMCStubbornSet.cpp [hunk 2]
```diff
@@ -32,8 +32,14 @@ namespace SMC {
...
-            assert(_interesting->get_negated() == false);
-            PQL::Visitor::visit(_interesting, q);
+            if(aphi){
+                assert(_interesting->get_negated() == false);
+                PQL::Visitor::visit(_interesting, q);
+            }
+            else{
+                assert(_interestingSMC->get_negated() == false);
+                PQL::Visitor::visit(_interestingSMC, q);
+            }
```
Comment (fulei345) [inline src/SMC/Stubborn/SMCStubbornSet.cpp]:
```suggestion
            if(aphi){
                assert(_interestingSMC->get_negated() == false);
                PQL::Visitor::visit(_interestingSMC, q);
            }
            else{
                assert(_interesting->get_negated() == false);
                PQL::Visitor::visit(_interesting, q);
            }
```

File: src/SMC/SMCMain.cpp [hunk 3]
```diff
@@ -72,7 +78,9 @@ namespace SMC
                 }
             }
 
-            stubset->prepare(&write);
+            if(aphi){
```
Comment (fulei345) [inline src/SMC/SMCMain.cpp]:
```suggestion
            if(!aphi){
```

**Review by fulei345 — CHANGES_REQUESTED** [review]

File: include/SMC/Stubborn/SMCStubbornSet.h [hunk 4]
```diff
@@ -19,26 +19,36 @@
...
+                _interestingSMC = std::make_unique<TVisitorSMC>(*this, _closure);
+                aphi = false;
         }
 
     private:
-        std::unique_ptr<PetriEngine::InterestingSMCTransitionVisitor> _interesting;
+        std::unique_ptr<PetriEngine::InterestingTransitionVisitor> _interesting;
+        std::unique_ptr<PetriEngine::InterestingSMCTransitionVisitor> _interestingSMC;
+
+        bool aphi = true;
```
Comment (fulei345) [inline include/SMC/Stubborn/SMCStubbornSet.h]:
```suggestion
        bool aphi = false;
```

File: include/SMC/Stubborn/SMCStubbornSet.h [hunk 5]
```diff
@@ -19,26 +19,36 @@
...
         void setInterestingVisitor()
         {
-            _interesting = std::make_unique<TVisitor>(*this, _closure);
+                _interesting = std::make_unique<TVisitor>(*this, _closure);
+        }
+        template <typename TVisitorSMC>
+        void setInterestingSMCVisitor()
+        {
+                _interestingSMC = std::make_unique<TVisitorSMC>(*this, _closure);
+                aphi = false;
```
Comment (fulei345) [inline include/SMC/Stubborn/SMCStubbornSet.h]:
```suggestion
                aphi = true;
```

**Review by fulei345 — CHANGES_REQUESTED** [review]

**Review by fulei345 — APPROVED** [review]

File: src/PetriEngine/options.cpp [hunk 6]
```diff
@@ -114,6 +114,10 @@ void printHelp() {
         "\n"
         "Options:\n"
         "  -smc <number of runs><depth>         Simulate a number of runs, with depth bound\n"
+        "  -i, <int transtions>                 Interesting Transitions:\n"
+        "                                       - SMCIT_ALL    Int trans for all trans\n"
+        "                                       - SMCIT_AM     Int trans for given marking\n"
+        "                                       - SMCIT_A      Int trans for a given p/n\n"
```
Comment (MikkelTJensen) [inline src/PetriEngine/options.cpp]: If this is some guidance stuff it should probably be "Interesting transitions" and not Int trans

Comment (fulei345) [reply src/PetriEngine/options.cpp]:
```suggestion
        "  -i, <Interesting transitions>                 Interesting transitions:\n"
        "                                       - SMCIT_ALL    Int trans for all trans\n"
        "                                       - SMCIT_AM     Int trans for given marking\n"
        "                                       - SMCIT_A      Int trans for a given p/n\n"
```

**Review by t-bsh — APPROVED** [review]
i assume it works

**Pull request merged**